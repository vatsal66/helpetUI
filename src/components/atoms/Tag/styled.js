import styled from 'styled-components';

export const StyledTag = styled.span`
  background-color: ${({ theme, color }) => theme.colors[color]};
  border-radius: 16px;
  color: ${({ theme }) => theme.colors.white};
  padding: 8px 48px;
`;
