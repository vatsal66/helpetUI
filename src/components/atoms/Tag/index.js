import React from 'react';
import PropTypes from 'prop-types';

import { StyledTag } from './styled';

const Tag = ({ color, children }) => {
  return <StyledTag color={color}>{children}</StyledTag>;
};

Tag.propTypes = {
  color: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export default Tag;
