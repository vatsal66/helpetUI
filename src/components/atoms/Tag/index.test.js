import React from 'react';

import renderWithTheme from '@mocks/renderWithTheme';

import Tag from './index';

describe('<Tag />', () => {
  const props = {
    color: 'veterinary',
  };

  it('should match snapshot file', () => {
    const { asFragment } = renderWithTheme(<Tag {...props}>Veterin√°rio</Tag>);
    expect(asFragment()).toMatchSnapshot();
  });
});
