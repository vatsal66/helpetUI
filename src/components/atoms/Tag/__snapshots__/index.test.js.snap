// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> should match snapshot file 1`] = `
<DocumentFragment>
  <span
    class="sc-AxjAm hSugdZ"
    color="veterinary"
  >
    Veterin√°rio
  </span>
</DocumentFragment>
`;
